lib/allure_report_publisher/lib/providers/gitlab.rb

```diff
@@ -62,24 +62,55 @@ def update_pr_description
       #
       # @return [void]
       def add_comment
-        unless comment
+        client.delete_merge_request_discussion_note(project, mr_iid, discussion.id, alert_comment.id) if alert_comment
+
+        if discussion
+          log_debug("Updating summary in comment with id #{discussion.id} in mr !#{mr_iid}")
+          client.update_merge_request_discussion_note(project, mr_iid, discussion.id, main_comment.id,
+                                                      body: url_section_builder.comment_body(main_comment.body))
+        else
           log_debug("Creating comment with summary for mr ! #{mr_iid}")
-          return client.create_merge_request_comment(project, mr_iid, url_section_builder.comment_body)
+          client.create_merge_request_comment(project, mr_iid, url_section_builder.comment_body)
         end
 
-        log_debug("Updating summary in comment with id #{comment.id} in mr !#{mr_iid}")
-        client.edit_merge_request_note(project, mr_iid, comment.id, url_section_builder.comment_body(comment.body))
+        return unless url_section_builder.summary_has_failures?
+
+        client.create_merge_request_discussion_note(project, mr_iid, discussion.id,
+                                                    body: alert_comment_text)
       end
 
-      # Existing comment with allure urls
+      # Existing discussion that has comment with allure urls
       #
       # @return [Gitlab::ObjectifiedHash]
-      def comment
-        client.merge_request_comments(project, mr_iid).auto_paginate.detect do |comment|
-          Helpers::UrlSectionBuilder.match?(comment.body)
+      def discussion
+        @discussion ||= client.merge_request_discussions(project, mr_iid).auto_paginate.detect do |discussion|
+          discussion.notes.any? { |note| Helpers::UrlSectionBuilder.match?(note.body) }
         end
       end
 
+      # Comment/note with allure urls
+      #
+      # @return [Gitlab::ObjectifiedHash]
+      def main_comment
+        @main_comment ||= discussion&.notes&.detect { |note| Helpers::UrlSectionBuilder.match?(note.body) }
+      end
+
+      # Comment with alert text
+      #
+      # @return [Gitlab::ObjectifiedHash]
+      def alert_comment
+        @alert_comment ||= discussion&.notes&.detect do |note|
+          note.body.include?(alert_comment_text)
+        end
+      end
+
+      # Text for alert comment
+      #
+      # @return [String]
+      def alert_comment_text
+        env("ALLURE_FAILURE_ALERT_COMMENT") || "There are some test failures that need attention"
```

Comment:
It's not very critical, but we should probably memoize this as well to not fetch from `ENV` multiple times